### PR TITLE
Label name editing

### PR DESCRIPTION
**Pull request by ActiveChooN**
### Motivation and context

Enable label name edition 

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
- [ ] I have added tests to cover my changes
~~- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
Hi, @dvkruchinin 

Please update tests (add this scenario) related with label changing

File: cvat/apps/engine/serializers.py [hunk 1]
```diff
@@ -84,12 +85,19 @@ def update_instance(validated_data, parent_instance):
...
-            logger.info("New {} label was created".format(db_label.name))
+        if not validated_data.get('id') is None:
+            db_label = models.Label.objects.get(id=validated_data['id'],
+                **instance)
+            db_label.name = validated_data.get('name', db_label.name)
         else:
-            logger.info("{} label was updated".format(db_label.name))
+            db_label = models.Label.objects.create(name=validated_data.get('name'), **instance)
+            logger.info("New {} label was created".format(db_label.name))
+        # (db_label, created) = models.Label.objects.get_or_create(name=validated_data['name'],
```
Comment (nmanovic) [inline cvat/apps/engine/serializers.py:95]: @ActiveChooN , do we need the comment? Should we keep logging code?

File: cvat/apps/engine/serializers.py [hunk 2]
```diff
@@ -84,12 +85,19 @@ def update_instance(validated_data, parent_instance):
         else:
             instance['task'] = parent_instance
             logger = slogger.task[parent_instance.id]
-        (db_label, created) = models.Label.objects.get_or_create(name=validated_data['name'],
-            **instance)
-        if created:
-            logger.info("New {} label was created".format(db_label.name))
+        if not validated_data.get('id') is None:
+            db_label = models.Label.objects.get(id=validated_data['id'],
```
Comment (nmanovic) [inline cvat/apps/engine/serializers.py:89]: @ActiveChooN , what if the id isn't valid? How does it handled?

**Comment by dvkruchinin** [comment]
> Please update tests (add this scenario) related with label changing

Sure. The test will be updated .

File: cvat-ui/src/components/labels-editor/label-form.tsx [hunk 3]
```diff
@@ -374,7 +374,6 @@ export default class LabelForm extends React.Component<Props> {
     private renderLabelNameInput(): JSX.Element {
         const { label, labelNames } = this.props;
         const value = label ? label.name : '';
```
Comment (vnishukov) [inline cvat-ui/src/components/labels-editor/label-form.tsx:376]: Is `const value = label?.name || ''` will works  here instead?

**Review by vnishukov — DISMISSED** [review]

File: cvat-ui/src/components/labels-editor/label-form.tsx (hunk 3 above)
Comment (ActiveChooN) [reply cvat-ui/src/components/labels-editor/label-form.tsx:376]: Guess it's quite the same thing

File: cvat/apps/engine/serializers.py (hunk 2 above)
Comment (ActiveChooN) [reply cvat/apps/engine/serializers.py]: Sure, added wrapper with 404 response. Will it be applicable, or just validationError will be enough, what do you think?

File: cvat/apps/engine/serializers.py (hunk 1 above)
Comment (ActiveChooN) [reply cvat/apps/engine/serializers.py]: Fixed

**Review by nmanovic — APPROVED** [review]

**Pull request merged**